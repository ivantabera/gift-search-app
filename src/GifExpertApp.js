import React, { useState } from 'react'
import { AddCategory } from './components/AddCategory'

export const GifExpertApp = () => {

    const [categories, setCategories] = useState(['One punch', 'Samurai X', 'Dragon ball'])

    //dos maneras de agregar un elemento al arreglo
    /* const handleAdd = () => {
        // setCategories([...categories, 'Bob Sponja']);
        setCategories( cate => [...categories, 'Bob sponja']);
    } */

    return (
        <>
            <h2>GifExpertApp</h2>
            <AddCategory setCategories={ setCategories } />

            <hr/>

            <ol>
                {
                    categories.map( (category) => {
                        return <li key={category}> {category} </li>
                    })
                }
            </ol>
        </>
    )
}

