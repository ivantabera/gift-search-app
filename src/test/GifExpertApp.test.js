import React from "react";
import '@testing-library/jest-dom';
import { shallow } from "enzyme";
import { GifExpertApp } from "../GifExpertApp"

describe('Pruebas en GifExpertApp', () => {

    test('Probar que el componente se renderice bien ', () => {
        
        const wrapper = shallow(<GifExpertApp />);
        expect(wrapper).toMatchSnapshot();
    })

    test('Mostrar una lista de categoria', () => {
        
        const categories = [ 'dragon ball', 'marvel' ];
        const wrapper = shallow(<GifExpertApp defaultCategories = {categories} />);

        expect(wrapper).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe(categories.length);


    })
    
    
})