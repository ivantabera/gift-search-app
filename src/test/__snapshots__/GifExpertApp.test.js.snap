// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifExpertApp Mostrar una lista de categoria 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="dragon ball"
      key="dragon ball"
    />
    <GifGrid
      category="marvel"
      key="marvel"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas en GifExpertApp Probar que el componente se renderice bien  1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
